src/gpu/nvidia/README.md

```diff
@@ -16,16 +16,47 @@ in-order queue when creating a stream if needed.
 
 ## Build command
 
+These build instructions have been validated with the latest CUDA 11.4 toolkit and DPC++ compiler release 2021.4.0.
```

Comment:
I replied in the thread as well. As for this PR, we implemented the workaround [here](https://github.com/oneapi-src/oneDNN/blob/3025f80fa59384c06c5ec2a3a4792e57be3f8107/cmake/SYCL.cmake#L68) so we expect that users will not have to deal with it anymore. You can validate it to make sure that it works for you without any adjustments on your side. If it works fine then we can close this PR.

Reply:
Thanks.  I've tried to install it again.  It appears that now icx and icpx are required to build using the packaged DPC++.  I am getting many warnings due to missing .so files like:

```
[855/1013] Linking CXX executable tests/gtests/test_iface_pd_buffer
/usr/bin/ld: warning: libsvml.so, needed by src/libdnnl.so.2.5, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libimf.so, needed by src/libdnnl.so.2.5, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libintlc.so.5, needed by src/libdnnl.so.2.5, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libirng.so, needed by /opt/intel/oneapi/compiler/2021.4.0/linux/bin/../lib/libsycl.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libsvml.so, needed by src/libdnnl.so.2.5, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libimf.so, needed by src/libdnnl.so.2.5, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libintlc.so.5, needed by src/libdnnl.so.2.5, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libirng.so, needed by /opt/intel/oneapi/compiler/2021.4.0/linux/bin/../lib/libsycl.so, not found (try using -rpath or -rpath-link)

```

Which is leading to all tests failing.  My cmake invocation is:

```
cmake -DCMAKE_C_COMPILER=/opt/intel/oneapi/compiler/2021.4.0/linux/bin/icx -DCMAKE_CXX_COMPILER=/opt/intel/oneapi/compiler/2021.4.0/linux/bin/icpx \
-DDNNL_CPU_RUNTIME=DPCPP -DDNNL_GPU_RUNTIME=DPCPP -DDNNL_GPU_VENDOR=NVIDIA \
-DCUBLAS_LIBRARY=/usr/local/cuda/lib64/libcublas.so \
-DCUBLAS_INCLUDE_DIR=/usr/local/cuda/include/ \
-DCUDNN_INCLUDE_DIR=/usr/local/cuda/include/ \
-G Ninja ..
```

Is this correct? What did you use for your build?

Thanks

Reply:
What compiler did you use? The DPC++ compiler with CUDA support is not part of binary distribution as far as I know. We only test open-source DPC++ compiler with CUDA support. The open-source compiler doesn't have icx/icpx drivers but only clang/clang++ which are allowed to be used for building oneDNN with  `DNNL_GPU_VENDOR=NVIDIA`.

Reply:
I used the packaged compiler from the Intel® oneAPI HPC Toolkit which I thought was the preferred way of building oneDNN according to this [comment](https://github.com/oneapi-src/oneDNN/pull/1172#discussion_r727369726)?  I had previously succeeded in building oneDNN using the packaged compiler using the instructions added in this pull request.

Reply:
I'm sorry if my comment was not clear enough.
I noticed that you were using `build` folder everywhere, while in the reality people do `make install` and get a single folder with standard folder tree inside: `bin`, `include`, `lib`, `share`, ... . That's why I had suggested to use this type of "package", but not a raw build from custom build folder since it has some artifacts that may be required for building oneDNN but they are not available when using this package.

None of distributed toolkits support cuda backend, this feature is available only for open-source version of compiler. Thanks.

Reply:
> I had previously succeeded in building oneDNN using the packaged compiler using the instructions added in this pull request

It's possible to build oneDNN with NVIDIA support using the compiler from Intel® oneAPI HPC Toolkit but it will fail at runtime due to lack of CUDA backend support in the compiler.

Reply:
> I'm sorry if my comment was not clear enough. I noticed that you were using `build` folder everywhere, while in the reality people do `make install` and get a single folder with standard folder tree inside: `bin`, `include`, `lib`, `share`, ... . That's why I had suggested to use this type of "package", but not a raw build from custom build folder since it has some artifacts that may be required for building oneDNN but they are not available when using this package.
> 
> None of distributed toolkits support cuda backend, this feature is available only for open-source version of compiler. Thanks.

Thanks for the clarification.  I've now confirmed that it builds and tests pass.  I've updated the build instructions accordingly.

Reply:
> > I had previously succeeded in building oneDNN using the packaged compiler using the instructions added in this pull request
> 
> It should be possible to build oneDNN with NVIDIA support using the compiler from Intel® oneAPI HPC Toolkit but it will fail at runtime due to lack of CUDA backend support in the compiler.

I did check that the tests passed for the case of the oneapi HPC Toolkit previously.  However since the updated cmake it no longer worked.  But so long as that is expected.

Reply:
Hi @densamoilov 
Is the updated doc in this PR OK now? I checked the instructions added in this PR with your new cmake patch.
Thanks

Reply:
Hi @JackAKirk, I removed dependency on OpenCL when CUDA backend is used (see fd51d4217bcf146678a90801615e2a3f1455cd6b). This means users don't have to manage OpenCL dependency when using oneDNN with NVIDIA support.

As for the OpenCL and Level Zero backends, it will be sufficient for user to simply specify a location of OpenCL via `OPENCLROOT`.

Reply:
So I think there is no need to add the clarifications to the doc anymore. If I'm not missing anything we can close this PR.
Thanks for helping us to find and fix all the issues.

Reply:
OK thanks